packages/core/src/app/utils/merkle-tree.ts

```diff
@@ -9,30 +9,19 @@ import {
   HashFunction,
   MerkleTreeInclusionProof,
   MerkleTreeNode,
+  ONE,
   SparseMerkleTree,
   TWO,
   ZERO,
 } from '../../types'
 import { keccak256 } from './crypto'
 
 /**
- * SparseMerkleTree implementation built using the optimizations implemented by Vitalik
- * here: https://github.com/ethereum/research/blob/master/sparse_merkle_tree/new_bintrie_optimized.py
- *
- * Namely, no intermediate nodes are persisted between a leaf node and its first ancestor
- * with a non-zero-hash sibling. In order to not store these intermediate nodes but
- * still make the leaf reachable through traversal, this ancestor node's value is
- * stored as its leaf node descendent's hash as well as the key that represents
- * the path to the leaf starting from this ancestor.
- *
- * To indicate that a node's value is a pointer to a leaf node instead of the typical
- * value of (leftChildHash + rightChildHash), the value is stored as 65 bytes instead
- * of the typical 64 bytes, with the first byte being a dummy byte.
+ * SparseMerkleTree implementation assuming a 256-bit hash algorithm is used.
  */
-export class OptimizedSparseMerkleTree implements SparseMerkleTree {
+export class SparseMerkleTreeImpl implements SparseMerkleTree {
   public static readonly emptyBuffer: Buffer = new Buffer(32).fill('\x00')
   private static readonly siblingBuffer: Buffer = new Buffer(1).fill('\x00')
```

Comment:
I think that these `new Buffer(...)`s can be replaced with `Buffer.alloc(32)` which will fill with zeros by default & is the new suggested way to allocate new buffers

Reply:
Done 👍